### PR TITLE
Sql relationship

### PR DESCRIPTION
**Pull request by Reem-Nosbaum**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -46,29 +47,31 @@ class Subjects(db.Model):
...
 	def get_dict(self):
 		return {
 			'id': self.id,
-			'user': self.user,
+			'author': self.author,
+			'user_id': self.user_id,
 			'subject': self.subject,
 			'body': self.body,
-			'group': self.group,
+			'chanel': self.chanel,
```
Comment (eladgunders) [inline app.py]: chanel => channel

File: app.py [hunk 2]
```diff
@@ -46,29 +47,31 @@ class Subjects(db.Model):
...
-	group = db.Column(db.Integer, db.ForeignKey('groups.id'), nullable=False)
+	chanel = db.Column(db.Integer, db.ForeignKey('channels.id'), nullable=False)
 	time_crated = db.Column(db.DateTime, nullable=False, default=datetime.now())
 	time_updated = db.Column(db.DateTime, nullable=True)
 
 	def get_dict(self):
 		return {
 			'id': self.id,
-			'user': self.user,
+			'author': self.author,
```
Comment (eladgunders) [inline app.py]: change to user

File: app.py [hunk 3]
```diff
@@ -46,29 +47,31 @@ class Subjects(db.Model):
...
 class Posts(db.Model):
 	__tablename__ = 'posts'
 	id = db.Column(db.Integer, primary_key=True)
-	user = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False)
+	author = db.relationship("Users", back_populates="user_posts")
+	user_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False)
 	subject = db.Column(db.Integer, db.ForeignKey('subjects.id'), nullable=False)
 	body = db.Column(db.Text, nullable=False)
-	group = db.Column(db.Integer, db.ForeignKey('groups.id'), nullable=False)
+	chanel = db.Column(db.Integer, db.ForeignKey('channels.id'), nullable=False)
```
Comment (eladgunders) [inline app.py]: chanel => channel

File: app.py [hunk 4]
```diff
@@ -46,29 +47,31 @@ class Subjects(db.Model):
...
 	id = db.Column(db.Integer, primary_key=True)
-	group = db.Column(db.Text, unique=True, nullable=False)
+	chanel = db.Column(db.Text, unique=True, nullable=False)
 
 
 class Posts(db.Model):
 	__tablename__ = 'posts'
 	id = db.Column(db.Integer, primary_key=True)
-	user = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False)
+	author = db.relationship("Users", back_populates="user_posts")
```
Comment (eladgunders) [inline app.py]: change to user

File: app.py [hunk 5]
```diff
@@ -46,29 +47,31 @@ class Subjects(db.Model):
 	subject = db.Column(db.Text, unique=True, nullable=False)
 
 
-class Groups(db.Model):
-	__tablename__ = 'groups'
+class Channels(db.Model):
+	__tablename__ = 'channels'
 	id = db.Column(db.Integer, primary_key=True)
-	group = db.Column(db.Text, unique=True, nullable=False)
+	chanel = db.Column(db.Text, unique=True, nullable=False)
```
Comment (eladgunders) [inline app.py]: chanel => channel

File: app.py [hunk 6]
```diff
@@ -29,6 +29,7 @@ class Users(db.Model):
 	password = db.Column(db.Text, nullable=False)
 	is_admin = db.Column(db.Boolean, nullable=False)
 	is_banned = db.Column(db.Boolean, nullable=False)
+	user_posts = db.relationship("Posts", back_populates="author")
```
Comment (eladgunders) [inline app.py]: why you need this relationship?

**Review by eladgunders — CHANGES_REQUESTED** [review]

File: app.py [hunk 7]
```diff
@@ -46,29 +46,31 @@ class Subjects(db.Model):
...
 	body = db.Column(db.Text, nullable=False)
-	group = db.Column(db.Integer, db.ForeignKey('groups.id'), nullable=False)
+	channel = db.Column(db.Integer, db.ForeignKey('channels.id'), nullable=False)
 	time_crated = db.Column(db.DateTime, nullable=False, default=datetime.now())
 	time_updated = db.Column(db.DateTime, nullable=True)
 
 	def get_dict(self):
 		return {
 			'id': self.id,
 			'user': self.user,
```
Comment (eladgunders) [inline app.py]: in the get dict why you need the user entity?

File: app.py [hunk 8]
```diff
@@ -46,29 +46,31 @@ class Subjects(db.Model):
...
 	id = db.Column(db.Integer, primary_key=True)
-	group = db.Column(db.Text, unique=True, nullable=False)
+	channel = db.Column(db.Text, unique=True, nullable=False)
 
 
 class Posts(db.Model):
 	__tablename__ = 'posts'
 	id = db.Column(db.Integer, primary_key=True)
-	user = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False)
+	user = db.relationship("Users", backref="users")
```
Comment (eladgunders) [inline app.py]: great, put it last(ufter time_updated) because uts not a column

**Review by eladgunders — CHANGES_REQUESTED** [review]

File: app.py (hunk 7 above)
Comment (Reem-Nosbaum) [reply app.py]: forgot to delate...

File: app.py (hunk 8 above)
Comment (Reem-Nosbaum) [reply app.py]: done 👌🏻

**Review by eladgunders — APPROVED** [review]

**Pull request merged**